src/lib/components/ui/Button/Button.jsx

```diff
@@ -85,37 +79,56 @@ const Button = (props) => {
       >
         {props.label}
       </span>
+      {props.afterLabel && (
+        <span
+          className={`
+            ${styles.afterLabel}
+            ${props.loading ? styles.loadingIcon : ''}
+          `.trim()}
+        >
+          {props.afterLabel}
+        </span>
+      )}
+      {props.endCorner && (
+        <span className={styles.endCorner}>
+          {props.endCorner}
+        </span>
+      )}
     </button>
   );
 };
 
 Button.defaultProps = {
+  afterLabel: null,
+  beforeLabel: null,
   block: false,
   clickHandler: null,
   disabled: false,
-  icon: null,
-  iconPosition: 'before',
+  endCorner: null,
   id: undefined,
   labelVisibility: 'all',
   loading: false,
   priority: 'default',
   size: 'medium',
+  startCorner: null,
   type: 'button',
   variant: 'primary',
 };
 
 Button.propTypes = {
+  afterLabel: PropTypes.element,
```

Comment:
Maybe we could rename it to `afterLabelIcon`? This goes for `beforeLabel` as well.

Reply:
This is intentional, it doesn't have to be icon – it can be badge or whatever.

Reply:
What about `afterLabelElement` then?

`afterLabel` just seems very cryptic…

Reply:
I was also considering that. I found inspiration in MUI which calls it—also unprecisely—`startIcon` and `endIcon`. Since I didn't want to (or maybe rather couldn't) restrict these elements to be just icons, I decided not to specify that at all. I can add the `Element` part to it though if you think it's better.